db/contentGraph.ts

```diff
@@ -46,58 +53,171 @@ export const getGrapherSlugs = (content: string | null): Set<string> => {
-                    })
-                } catch (err) {
-                    // ConflictErrors occur here when a chart <-> post
-                    // relationship already exists
-                    if (!(err instanceof ConflictError)) {
-                        throw err
-                    }
-                }
+                // ConflictErrors occur here when a chart <-> post
+                // relationship already exists
+                throwAllButConflictError(err)
             }
         }
     }
-    return store
+}
+
+export const removeUnpublishedDocuments = async (graph: any): Promise<void> => {
+    const ids = (
+        await graph.find(GraphType.Document, null, {
+            fields: { id: true },
+            match: { title: null },
+        })
+    ).payload.records?.map((record: { id: number }) => record.id)
+
+    if (isEmpty(ids)) return
+    await graph.delete(GraphType.Document, ids)
+}
+
+export const fortuneRecordTypes = {
+    [GraphType.Document]: {
+        title: String,
+        slug: String,
+        type: String,
+        content: String,
+        image: String,
+        parentTopics: [Array(GraphType.Document), "childrenTopics"],
+        childrenTopics: [Array(GraphType.Document), "parentTopics"],
+        embeddedCharts: [Array(GraphType.Chart), "embeddedIn"],
+        charts: [Array(GraphType.Chart), "parentTopics"],
+    },
+    [GraphType.Chart]: {
+        title: String,
+        slug: String,
+        type: String,
+        embeddedIn: [Array(GraphType.Document), "embeddedCharts"],
+        parentTopics: [Array(GraphType.Document), "charts"],
+    },
+}
+
+export const getContentGraph = once(async () => {
+    const graph = fortune(fortuneRecordTypes, {
+        adapter: [
+            MemoryAdapter,
+            {
+                // see https://github.com/fortunejs/fortune/commit/70593721efae304ff2db40d1b8f9b43295fed79b#diff-ebb028a2d1528eac83ee833036ef6e50bed6fb7b2b7137f59ac1fb567a5e6ec2R25
+                recordsPerType: Infinity,
+            },
+        ],
+    })
+
+    // Add documents (topics, articles)
+    const orderBy = "orderby:{field:MODIFIED, order:DESC}"
+    const entries = await getDocumentsInfo(
+        WP_PostType.Page,
+        "",
+        `categoryId: ${ENTRIES_CATEGORY_ID}, ${orderBy}`
+    )
+    const posts = await getDocumentsInfo(WP_PostType.Post, "", orderBy)
+    const documents = [...entries, ...posts]
+    await addDocumentsToGraph(documents, graph)
+
+    // Add all public charts
+    const allCharts = await getChartsRecords()
+    await addChartsToGraph(allCharts, graph)
+
+    // Add all embedded charts
+    await addEmbeddedChartsToGraph(documents, graph)
+
+    // Unpublished topics might be referenced as a parent. As a result, only the
+    // referential information (id) is present.
+    await removeUnpublishedDocuments(graph)
+
+    return graph
 })
+
+const main = async (): Promise<void> => {
+    const graph = await getContentGraph()
```

Comment:
This does nothing? (Was it used for testing, and then rewritten?)

Reply:
yep I use it for testing (through the "Build content graph" launch.json task). We could do the same with  `indexContentGraphToAlgolia()` which also calls `getContentGraph()` but I find using that one a bit nerve-racking given the potential for burning through our indexing quota. I'll comment out this call to make it more obvious but let me know if you have a better idea.